src/datasource.ts

```diff
@@ -805,23 +805,25 @@ export default class IrondbDatasource extends DataSourceApi<IrondbQueryInterface
             });
     }
 
-    throwerr(err) {
-        log(() => 'throwerr() err = ' + err);
-        if (err.data && err.data.error) {
-            throw new Error('Circonus IRONdb Error: ' + err.data.error);
-        } else if (err.data && err.data.user_error) {
-            const name = err.data.method || 'IRONdb';
-            let suffix = '';
-            if (err.data.user_error.query) {
-                suffix = ' in"' + err.data.user_error.query + '"';
+    throwerr(error: any) {
+        log(() => 'throwerr() err = ' + JSON.stringify(error));
+        if (typeof error !== 'string') {
+            if (error.data) {
+                if (error.data.message) {
+                    let message = JSON.parse(error.data.message);
```

Comment:
JSON parsing should be done inside a try/catch block to avoid parsing errors.
let message;
try {
    message = JSON.parse(error.data.message);
}
catch (error) {
    //throw JSON error here
}

Reply:
Added error handling as requested.